File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def check(string):
    stack = []
    for i in string:
        if i in ["(", "[", "{"]:
            stack.append(i)
        else:
            if len(stack) == 0:
                return 0
            top = stack.pop() 
            if top+i not in ["()", "[]", "{}"]:
                return 0
    if len(stack) > 0:
        return 0
    return 1

def solution(s):
    answer = 0
    n = len(s)
    for i in range(n):
        answer += check(s[i:] + s[:i])
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

bool is_correct(string s) {
    
    stack<char> st;
    for (int i = 0; i < s.size(); i++) {
        char c = s[i];
        if (c == '(' || c == '[' || c == '{') {
            st.push(c);
        } else {
            if (st.empty()) {
                return false;
            }
            if (c == ')') {
                if (st.top() != '(') {
                    return false;
                }
                st.pop();
            } else if (c == ']') {
                if (st.top() != '[') {
                    return false;
                }
                st.pop();
            } else {
                if (st.top() != '{') {
                    return false;
                }
                st.pop();
            }
        }
    }
    return st.empty();
    
}

int solution(string s) {
    int answer = 0;
    
    int pointer = 0;
    while (pointer < s.size()) {
        answer += is_correct(s.substr(pointer) + s.substr(0, pointer));
        pointer++;
    }
    
    return answer;
}